fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -354,15 +354,15 @@ public Response deleteObject() {
                 METHOD_NOT_ALLOWED);
         }
 
-        evaluateRequestPreconditions(request, servletResponse, resource(), session);
+        evaluateRequestPreconditions(request, servletResponse, resource(), transaction);
 
         LOGGER.info("Delete resource '{}'", externalPath);
 
         final AcquiredLock lock = lockManager.lockForDelete(resource().getPath());
 
         try {
             resource().delete();
-            session.commit();
+            transaction.commit();
```

Comment:
I don't think we want to commit transactions in these restful methods unless transaction.isShortlived() is true. Otherwise, committing should happen on explicit commits from the client.

Reply:
+1

Reply:
Done